x-pack/plugins/apm/common/utils/get_preferred_bucket_size_and_data_source.test.ts

```diff
@@ -4,48 +4,55 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-import { ApmDataSource } from '../data_source';
+import { ApmDataSourceWithSummary } from '../data_source';
 import { ApmDocumentType } from '../document_type';
 import { RollupInterval } from '../rollup';
 import {
   getPreferredBucketSizeAndDataSource,
   intervalToSeconds,
 } from './get_preferred_bucket_size_and_data_source';
 
-const serviceTransactionMetricSources: ApmDataSource[] = [
+const serviceTransactionMetricSources: ApmDataSourceWithSummary[] = [
```

Comment:
why is `hasDurationSummaryField` here but `hasDocs` isn't? I guess that `hasDocs` is implied to be true always, otherwise it wouldn't be passed to the function, but maybe that's an inconsistency that we need to fix.